packages/sdk-drop-in/tests/iframe.test.ts

```diff
@@ -0,0 +1,124 @@
+import { WidgetType } from '@/apps/MoniteApp';
+import { test, expect } from '@playwright/test';
+
+import { writeFileSync } from 'fs';
+import * as process from 'node:process';
+import * as path from 'path';
+import { fileURLToPath } from 'url';
+
+const mockedConfigData = {
+  stand: 'dev',
+  api_url: 'https://api.dev.monite.com',
+  app_basename: 'monite-iframe-app',
+  app_hostname: '127.0.0.1',
+  entity_user_id: 'mocked_entity_id',
+  client_id: 'mocked_client_id',
+  client_secret: 'mocked_client_secret',
+};
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+if (process.env.CI) {
+  const configFilePath = path.join(__dirname, '../public', 'config.json');
+  writeFileSync(configFilePath, JSON.stringify(mockedConfigData, null, 2));
+}
+
+const consumerPage = '/monite-iframe-app-demo';
+
+const routingPaths: Record<WidgetType, string> = {
+  payables: '/payables',
+  receivables: '/receivables',
+  counterparts: '/counterparts',
+  products: '/products',
+  tags: '/tags',
+  'approval-policies': '/approval-policies',
+  onboarding: '/onboarding',
+};
+
+test.beforeEach(async ({ page }) => {
+  await page.route('/config.json', async (route) => {
+    if (process.env.CI) {
+      await route.fulfill({
+        contentType: 'application/json',
+        body: process.env.MONITE_E2E_APP_ADMIN_CONFIG_JSON,
+      });
+    } else {
+      await route.continue();
+    }
+  });
+
+  await page.goto(`${consumerPage}${routingPaths.receivables}`, {
+    waitUntil: 'domcontentloaded',
+    timeout: 60000,
+  });
+
+  const iframeElement = await page.locator('iframe').elementHandle();
+
+  if (!iframeElement) {
+    await page.reload({ waitUntil: 'domcontentloaded', timeout: 60000 });
+  }
+
+  const frame = await iframeElement?.contentFrame();
+
+  if (!frame) {
+    await page.reload({ waitUntil: 'domcontentloaded', timeout: 60000 });
+  }
+
+  await frame?.waitForURL(new RegExp('.*/receivables.*', 'i'), {
+    timeout: 60000,
+  });
+});
```

Comment:
I'd like to ask you not to add test logic to `test.beforeEach(...)`. `test.beforeEach(...)` is the place for mocks, but not for implementing the business logic of the test.

In your case, if I move the test "__test the theme switcher__" to another file, it will fail with a timeout. Although it requires a simple `await page.goto(...)`, without which the test becomes incomprehensible.

Sorry, I didn't pay attention to this last time.